packages/mui-material/src/Grid/Grid.js

```diff
@@ -315,24 +331,44 @@ const Grid = React.forwardRef(function Grid(inProps, ref) {
   // columns set with default breakpoint unit of 12
   const columns = container ? columnsProp || 12 : columnsContext;
 
+  const breakpointsDefaultValues = {};
+  const breakpointsProps = {};
+  const propsRestFiltered = {};
+
+  breakpoints.keys.forEach((key) => {
+    breakpointsDefaultValues[key] = false;
+  });
+
+  Object.keys(propsRest).forEach((key) => {
```

Comment:
Instead of navigating through the rest of the props (where we do not know how many items there may be), how about we navigate through the breakpoint values (which should be a small set of values), and remove those from the rest of the props?